src/encodings/encoding.rs

```diff
@@ -351,6 +354,68 @@ impl<T: DataType> Encoder<T> for DictEncoder<T> {
   }
 }
 
+// ----------------------------------------------------------------------
+// RLE encoding
+
+const DEFAULT_RLE_BUFFER_LEN: usize = 1024;
+
+/// RLE encoding for values.
+/// Currently is used only for data pages v2 and supports boolean types.
+pub struct RleValueEncoder<T: DataType> {
```

Comment:
Seems this may also encode the input data with bit-packed encoding? if so, we may want to clarify in the comments.

Reply:
I added the note that it is a hybrid encoding (bit packed + rle).